core/tests/Test.Microsoft.Identity.Core.Unit/PlatformProxyFactoryTests.cs

```diff
@@ -0,0 +1,50 @@
+﻿//----------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+using Microsoft.Identity.Core;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Test.Microsoft.Identity.Core.Unit
+{
+    [TestClass]
+    public class PlatformProxyFactoryTests
+    {
+        [TestMethod]
+        public void PlatformProxyFactoryCachesTheProxy()
```

Comment:
>PlatformProxyFactoryCachesTheProxy [](start = 20, length = 34)

imo, we need an object which holds the proxy during the lifetime of the client applications. (perhaps something setup in the module initilizer).

Reply:
Made it a singleton. It doesn't need to be initialized in the module initializer because there is no library specific logic. 

---
In reply to: [221832547](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1284#discussion_r221832547) [](ancestors = 221832547)